examples/opentracing-shim/tracer.js

```diff
@@ -0,0 +1,24 @@
+const { NodeTracer } = require("@opentelemetry/node");
+const { SimpleSpanProcessor } = require("@opentelemetry/tracing");
+const { JaegerExporter } = require("@opentelemetry/exporter-jaeger");
+const { ZipkinExporter } = require("@opentelemetry/exporter-zipkin");
+const { TracerShim } = require("@opentelemetry/shim-opentracing");
+
+const opentracing = require("opentracing");
+
+function init(serviceName) {
```

Comment:
The fact that this file does OpenTracing and `OpenTelemetry` things together might be confusing, especially since the names are so similar. I would rather have one file that sets up open tracing the way you would normally set it up with open tracing, and another file called something like `shim.js` that sets up the shim. This way the example more clearly shows where the shim happens.

```javascript
// shim.js

exports.shim = function shim(serviceName) {
  const tracer = new NodeTracer();
  tracer.addSpanProcessor(new SimpleSpanProcessor(getExporter()));

  return new TracerShim(tracer);
}

function getExporter() {
  const type = process.env.EXPORTER || "jaeger";
  if (type.match(/^z/)) {
    return new ZipkinExporter({ serviceName });
  }
  return new JaegerExporter({ serviceName, flushInterval: 100 });
}
```

```javascript
// client.js
const shim = require("./shim").shim("http_client_service");

opentracing.initGlobalTracer(shim);
```

Reply:
I was initially imagining people usually have a tracer package which does initialization for open tracing and vendor specific instance (e.g. Zipkin or Jaeger). In this case, open telemetry is just another vendor so I made this decision to put all of them together.

I'm wondering if this makes sense to you and if adding some comments in the code can help or it will still be better to be separated as you suggest?

Many thanks for the suggestion! 👍

Reply:
I was initially imagining people usually have a tracer package which does initialization for open tracing and vendor specific instance (e.g. Zipkin or Jaeger). In this case, open telemetry is just another vendor so I made this decision to put all of them together.

I'm wondering if this makes sense to you and if adding some comments in the code can help or it will still be better to be separated as you suggest?

Many thanks for the suggestion! 👍

Reply:
I was initially imagining people usually have a tracer package which does initialization for open tracing and vendor specific instance (e.g. Zipkin or Jaeger). In this case, open telemetry is just another vendor so I made this decision to put all of them together.

I'm wondering if this makes sense to you and if adding some comments in the code can help or it will still be better to be separated as you suggest?

Many thanks for the suggestion! 👍

Reply:
I think i can go either way. I would still rather have the shim separate from the opentracing tracer so it is clear where the break is

Reply:
Sure, as your suggestion, I've created a `shim` file to make thing more clear. Thanks!